src/javascript/javascript-parser.ts

```diff
@@ -92,45 +102,56 @@ export function parseJs(
     contents: string,
     file: string,
     locationOffset?: LocationOffset,
-    warningCode?: string): ParseResult {
+    warningCode?: string,
+    sourceType?: SourceType): ParseResult {
   if (!warningCode) {
     warningCode = 'parse-error';
   }
-  const options = Object.assign(
-      {sourceType: 'script' as ('script' | 'module')}, baseParseOptions);
+
+  let program: estree.Program;
+
   try {
+    if (!sourceType) {
+      try {
+        sourceType = 'script';
+        program = espree.parse(
+            contents,
+            Object.assign(
+                {sourceType: 'script' as SourceType}, baseParseOptions));
+      } catch (_ignored) {
+        sourceType = 'module';
```

Comment:
I wonder if this should check for a `SyntaxError`. @rictic why didn't we check before? Can we add a comment for posterity now?